Fix test_linesearch_vector_bound_enforcement failure under win32

### Summary of Issue

The `test_linesearch_vector_bound_enforcement` check fails on GitHub Actions only for win32.

### Issue Type

- [X] Bug
- [ ] Enhancement  
      POEM #___ <N/A or POEM number at https://github.com/OpenMDAO/POEMs>

### Example

```
Traceback (most recent call last):
  File "C:\Miniconda\envs\test\lib\site-packages\testflo\test.py", line 417, in _try_call
    func()
  File "C:\Miniconda\envs\test\lib\site-packages\openmdao\solvers\linesearch\tests\test_backtracking.py", line 447, in test_linesearch_vector_bound_enforcement
    top.run_model()
  File "C:\Miniconda\envs\test\lib\contextlib.py", line 120, in __exit__
    next(self.gen)
  File "C:\Miniconda\envs\test\lib\site-packages\openmdao\utils\assert_utils.py", line 62, in assert_warning
    raise AssertionError(msg)
AssertionError: Did not see expected SolverWarning:
'comp.z' exceeds lower bounds
  Val: [1.33333333 1.33333333 1.33333333]
  Lower: [1.5 1.5 1.5]
```

### Environment

Operating System: GitHub Actions win32 platform

This is not just a Windows issue.  Also seen now on CentOS Linux with environment shown below:
```
/mdao/u/swryan/benchmark/runner/openmdao_repos/OpenMDAO/openmdao/solvers/linesearch/tests/test_backtracking.py:TestBoundsEnforceLSArrayBounds.test_linesearch_vector_bound_enforcement ... FAIL (00:00:0.05, 608 MB)
/mdao/u/swryan/benchmark/runner/openmdao_repos/OpenMDAO/openmdao/solvers/linesearch/backtracking.py:39: SolverWarning:'comp.z' exceeds lower bounds
  Val: [1.33333333333 1.33333333333 1.33333333333]
  Lower: [1.5 1.5 1.5]

[... LINES DELETED ...]

Traceback (most recent call last):
  File "/mdao/u/swryan/.conda/envs/openmdao_20211005-005514/lib/python3.6/site-packages/testflo/test.py", line 417, in _try_call
    func()
  File "/mdao/u/swryan/benchmark/runner/openmdao_repos/OpenMDAO/openmdao/solvers/linesearch/tests/test_backtracking.py", line 450, in test_linesearch_vector_bound_enforcement
    top.run_model()
  File "/mdao/u/swryan/.conda/envs/openmdao_20211005-005514/lib/python3.6/contextlib.py", line 88, in __exit__
    next(self.gen)
  File "/mdao/u/swryan/benchmark/runner/openmdao_repos/OpenMDAO/openmdao/utils/assert_utils.py", line 62, in assert_warning
    raise AssertionError(msg)
AssertionError: Did not see expected SolverWarning:
'comp.z' exceeds lower bounds
  Val: [1.33333333 1.33333333 1.33333333]
  Lower: [1.5 1.5 1.5]
```
```
# packages in environment at /mdao/u/swryan/.conda/envs/openmdao_20211005-005514:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main
_openmp_mutex             4.5                       1_gnu
aiounittest               1.4.0                    pypi_0    pypi
alabaster                 0.7.12                   pypi_0    pypi
anyio                     3.3.2                    pypi_0    pypi
argon2-cffi               21.1.0                   pypi_0    pypi
async-generator           1.10                     pypi_0    pypi
attrs                     20.3.0                   pypi_0    pypi
babel                     2.9.1                    pypi_0    pypi
backcall                  0.2.0                    pypi_0    pypi
beautifulsoup4            4.10.0                   pypi_0    pypi
blas                      1.0                         mkl
bleach                    4.1.0                    pypi_0    pypi
bokeh                     2.3.3                    pypi_0    pypi
brotli                    1.0.9                he6710b0_2
bzip2                     1.0.8                h7b6447c_0
c-ares                    1.17.1               h27cfd23_0
ca-certificates           2021.5.30            ha878542_0    conda-forge
certifi                   2021.5.30        py36h5fab9bb_0    conda-forge
cffi                      1.14.6                   pypi_0    pypi
charset-normalizer        2.0.6                    pypi_0    pypi
click                     7.1.2                    pypi_0    pypi
colorama                  0.4.4                    pypi_0    pypi
contextvars               2.4                      pypi_0    pypi
coverage                  4.5.4                    pypi_0    pypi
curl                      7.78.0               h1ccaba5_0
cycler                    0.10.0                     py_2    conda-forge
cython                    0.29.24          py36hc4f0c31_0    conda-forge
dataclasses               0.8                      pypi_0    pypi
dbus                      1.13.18              hb2f20db_0
decorator                 4.4.2                    pypi_0    pypi
defusedxml                0.7.1                    pypi_0    pypi
docutils                  0.16                     pypi_0    pypi
entrypoints               0.3                      pypi_0    pypi
expat                     2.4.1                h2531618_2
fontconfig                2.13.1               h6c09931_0
fonttools                 4.25.0             pyhd3eb1b0_0
freetype                  2.10.4               h5ab3b9f_0
gettext                   0.21.0               hf68c758_0
git                       2.32.0          pl5262hc120c5b_1
gitdb                     4.0.7                    pypi_0    pypi
gitpython                 3.1.18                   pypi_0    pypi
glib                      2.69.1               h5202010_0
greenlet                  1.1.2                    pypi_0    pypi
gst-plugins-base          1.14.0               h8213a91_2
gstreamer                 1.14.0               h28cd5cc_2
hypre                     2.15.1            hc98498a_1002    conda-forge
icu                       58.2                 he6710b0_3
idna                      3.2                      pypi_0    pypi
imagesize                 1.2.0                    pypi_0    pypi
immutables                0.16                     pypi_0    pypi
importlib-metadata        4.8.1                    pypi_0    pypi
importlib-resources       3.0.0                    pypi_0    pypi
intel-openmp              2021.3.0          h06a4308_3350
ipykernel                 5.5.5                    pypi_0    pypi
ipyparallel               6.3.0                    pypi_0    pypi
ipython                   7.16.1                   pypi_0    pypi
ipython-genutils          0.2.0                    pypi_0    pypi
ipywidgets                7.6.5                    pypi_0    pypi
jedi                      0.18.0                   pypi_0    pypi
jinja2                    3.0.2                    pypi_0    pypi
jpeg                      9d                   h7f8727e_0
jsonschema                3.2.0                    pypi_0    pypi
jupyter                   1.0.0                    pypi_0    pypi
jupyter-book              0.11.3                   pypi_0    pypi
jupyter-cache             0.4.3                    pypi_0    pypi
jupyter-client            7.0.5                    pypi_0    pypi
jupyter-console           6.4.0                    pypi_0    pypi
jupyter-core              4.8.1                    pypi_0    pypi
jupyter-server            1.11.1                   pypi_0    pypi
jupyter-server-mathjax    0.2.3                    pypi_0    pypi
jupyter-sphinx            0.3.1                    pypi_0    pypi
jupyterlab-widgets        1.0.2                    pypi_0    pypi
jupytext                  1.10.3                   pypi_0    pypi
kiwisolver                1.3.1            py36h605e78d_1    conda-forge
krb5                      1.19.2               hac12032_0
latexcodec                2.0.1                    pypi_0    pypi
lcms2                     2.12                 h3be6417_0
ld_impl_linux-64          2.35.1               h7274673_9
libblas                   3.9.0            11_linux64_mkl    conda-forge
libcblas                  3.9.0            11_linux64_mkl    conda-forge
libcurl                   7.78.0               h0b77cf5_0
libedit                   3.1.20210714         h7f8727e_0
libev                     4.33                 h7f8727e_1
libffi                    3.3                  he6710b0_2
libgcc-ng                 9.3.0               h5101ec6_17
libgfortran-ng            7.5.0               ha8ba4b0_17
libgfortran4              7.5.0               ha8ba4b0_17
libgomp                   9.3.0               h5101ec6_17
libiconv                  1.15                 h63c8f33_5
liblapack                 3.9.0            11_linux64_mkl    conda-forge
libnghttp2                1.41.0               hf8bcb03_2
libpng                    1.6.37               hbc83047_0
libssh2                   1.9.0                h1ba5d50_1
libstdcxx-ng              9.3.0               hd4cf53a_17
libtiff                   4.2.0                h85742a9_0
libuuid                   1.0.3                h1bed415_2
libwebp-base              1.2.0                h27cfd23_0
libxcb                    1.14                 h7b6447c_0
libxml2                   2.9.12               h03d6c58_0
linkify-it-py             1.0.1                    pypi_0    pypi
lz4-c                     1.9.3                h295c915_1
markdown-it-py            0.6.2                    pypi_0    pypi
markupsafe                2.0.1                    pypi_0    pypi
matplotlib                3.2.2                         1    conda-forge
matplotlib-base           3.2.2            py36hef1b27d_0
mdit-py-plugins           0.2.6                    pypi_0    pypi
metis                     5.1.0             h58526e2_1006    conda-forge
mistune                   0.8.4                    pypi_0    pypi
mkl                       2021.3.0           h06a4308_520
mpi                       1.0                       mpich    conda-forge
mpi4py                    3.0.0           py36_mpich_1003    conda-forge
mpich                     3.2.1             hc99cbb1_1014    conda-forge
mumps-include             5.2.1                         4    conda-forge
mumps-mpi                 5.2.1                h645f208_4    conda-forge
munkres                   1.1.4                      py_0
myst-nb                   0.12.0                   pypi_0    pypi
myst-parser               0.13.7                   pypi_0    pypi
nbclient                  0.5.4                    pypi_0    pypi
nbconvert                 5.6.1                    pypi_0    pypi
nbdime                    3.1.0                    pypi_0    pypi
nbformat                  5.1.3                    pypi_0    pypi
ncurses                   6.2                  he6710b0_1
nest-asyncio              1.5.1                    pypi_0    pypi
networkx                  2.5.1                    pypi_0    pypi
notebook                  6.4.4                    pypi_0    pypi
numpy                     1.19.5           py36hfc0c790_2    conda-forge
numpydoc                  1.1.0                    pypi_0    pypi
objgraph                  3.5.0                    pypi_0    pypi
olefile                   0.46               pyhd3eb1b0_0
openjpeg                  2.4.0                h3ad879b_0
openmdao                  3.12.1.dev0               dev_0    <develop>
openssl                   1.1.1k               h7f98852_0    conda-forge
packaging                 21.0                     pypi_0    pypi
pandocfilters             1.5.0                    pypi_0    pypi
parameterized             0.8.1                    pypi_0    pypi
parmetis                  4.0.3             h9f7b9cf_1003    conda-forge
parso                     0.8.2                    pypi_0    pypi
pcre                      8.45                 h295c915_0
pcre2                     10.35                h14c3975_1
perl                      5.26.2               h14c3975_0
petsc                     3.11.4               h6322da9_1    conda-forge
petsc4py                  3.11.0           py36h3d81f39_1    conda-forge
pexpect                   4.8.0                    pypi_0    pypi
pickleshare               0.7.5                    pypi_0    pypi
pillow                    8.3.2                    pypi_0    pypi
pip                       21.2.4             pyhd8ed1ab_0    conda-forge
prometheus-client         0.11.0                   pypi_0    pypi
prompt-toolkit            3.0.20                   pypi_0    pypi
psutil                    5.8.0            py36h8f6f2f9_1    conda-forge
ptscotch                  6.0.8                h2e95eaa_0    conda-forge
ptyprocess                0.7.0                    pypi_0    pypi
pybtex                    0.24.0                   pypi_0    pypi
pybtex-docutils           1.0.1                    pypi_0    pypi
pycodestyle               2.7.0                    pypi_0    pypi
pycparser                 2.20                     pypi_0    pypi
pydata-sphinx-theme       0.6.3                    pypi_0    pypi
pydocstyle                2.0.0                    pypi_0    pypi
pydoe2                    1.3.0                    pypi_0    pypi
pygments                  2.10.0                   pypi_0    pypi
pyoptsparse               2.1.5                    pypi_0    pypi
pyparsing                 2.4.7              pyhd3eb1b0_0
pyrsistent                0.18.0                   pypi_0    pypi
python                    3.6.13          hffdb5ce_0_cpython    conda-forge
python-dateutil           2.8.2              pyhd3eb1b0_0
python-graphviz           0.17                     pypi_0    pypi
python_abi                3.6                     2_cp36m    conda-forge
pytz                      2021.3                   pypi_0    pypi
pyyaml                    5.4.1                    pypi_0    pypi
pyzmq                     22.3.0                   pypi_0    pypi
qt                        5.9.7                h5867ecd_1
qtconsole                 5.1.1                    pypi_0    pypi
qtpy                      1.11.2                   pypi_0    pypi
readline                  8.1                  h27cfd23_0
requests                  2.26.0                   pypi_0    pypi
requests-unixsocket       0.2.0                    pypi_0    pypi
scalapack                 2.0.2             hfacbc1e_1006    conda-forge
scipy                     1.5.3            py36h976291a_0    conda-forge
scotch                    6.0.8                h0eec0ba_1    conda-forge
send2trash                1.8.0                    pypi_0    pypi
setuptools                58.0.4           py36h5fab9bb_2    conda-forge
six                       1.16.0             pyhd3eb1b0_0
smmap                     4.0.0                    pypi_0    pypi
sniffio                   1.2.0                    pypi_0    pypi
snowballstemmer           2.1.0                    pypi_0    pypi
soupsieve                 2.2.1                    pypi_0    pypi
sphinx                    3.5.4                    pypi_0    pypi
sphinx-book-theme         0.1.5                    pypi_0    pypi
sphinx-comments           0.0.3                    pypi_0    pypi
sphinx-copybutton         0.4.0                    pypi_0    pypi
sphinx-external-toc       0.2.3                    pypi_0    pypi
sphinx-jupyterbook-latex  0.4.3                    pypi_0    pypi
sphinx-multitoc-numbering 0.1.3                    pypi_0    pypi
sphinx-panels             0.5.2                    pypi_0    pypi
sphinx-sitemap            2.2.0                    pypi_0    pypi
sphinx-thebe              0.0.10                   pypi_0    pypi
sphinx-togglebutton       0.2.3                    pypi_0    pypi
sphinxcontrib-applehelp   1.0.2                    pypi_0    pypi
sphinxcontrib-bibtex      2.2.1                    pypi_0    pypi
sphinxcontrib-devhelp     1.0.2                    pypi_0    pypi
sphinxcontrib-htmlhelp    2.0.0                    pypi_0    pypi
sphinxcontrib-jsmath      1.0.1                    pypi_0    pypi
sphinxcontrib-qthelp      1.0.3                    pypi_0    pypi
sphinxcontrib-serializinghtml 1.1.5                    pypi_0    pypi
sqlalchemy                1.4.25                   pypi_0    pypi
sqlite                    3.36.0               hc218d9a_0
sqlitedict                1.7.0                    pypi_0    pypi
suitesparse               4.5.6             h717dc36_1204    conda-forge
swig                      4.0.2                h2531618_3
tabulate                  0.8.9                    pypi_0    pypi
tbb                       2020.2               h4bd325d_4    conda-forge
terminado                 0.12.1                   pypi_0    pypi
testflo                   1.4.5.1                  pypi_0    pypi
testpath                  0.5.0                    pypi_0    pypi
tk                        8.6.11               h1ccaba5_0
toml                      0.10.2                   pypi_0    pypi
tornado                   6.1              py36h8f6f2f9_1    conda-forge
traitlets                 4.3.3                    pypi_0    pypi
typing-extensions         3.10.0.2                 pypi_0    pypi
tzdata                    2021a                h5d7bf9c_0
uc-micro-py               1.0.1                    pypi_0    pypi
urllib3                   1.26.7                   pypi_0    pypi
wcwidth                   0.2.5                    pypi_0    pypi
webencodings              0.5.1                    pypi_0    pypi
websocket-client          1.2.1                    pypi_0    pypi
wheel                     0.37.0             pyhd3eb1b0_1
widgetsnbextension        3.5.1                    pypi_0    pypi
wrapt                     1.13.1                   pypi_0    pypi
xz                        5.2.5                h7b6447c_0
zipp                      3.6.0                    pypi_0    pypi
zlib                      1.2.11               h7b6447c_3
zstd                      1.4.9                haebb681_0
```

--- a/openmdao/solvers/linesearch/tests/test_backtracking.py
+++ b/openmdao/solvers/linesearch/tests/test_backtracking.py
@@ -15,6 +15,7 @@ from openmdao.test_suite.components.implicit_newton_linesearch \
     import ImplCompTwoStates, ImplCompTwoStatesArrays
 from openmdao.test_suite.components.sellar import SellarDis1, SellarDis2withDerivatives
 from openmdao.utils.assert_utils import assert_near_equal, assert_warning
+from openmdao.utils.general_utils import printoptions
 
 
 class TestArmejoGoldsteinBounds(unittest.TestCase):
@@ -426,8 +427,6 @@ class TestBoundsEnforceLSArrayBounds(unittest.TestCase):
         self.top = top
         self.ub = np.array([2.6, 2.5, 2.65])
 
-    @unittest.skipIf((os.environ.get("GITHUB_ACTION") and sys.platform == 'win32'), \
-        'Lower bounds check fails on Windows.')
     def test_linesearch_vector_bound_enforcement(self):
         top = self.top
 
@@ -445,9 +444,10 @@ class TestBoundsEnforceLSArrayBounds(unittest.TestCase):
         for ind in range(3):
             assert_near_equal(top['comp.z'][ind], [1.5], 1e-8)
 
-        msg = (f"'comp.z' exceeds lower bounds\n  Val: [1.33333333 1.33333333 1.33333333]\n  Lower: [1.5 1.5 1.5]\n")
-        with assert_warning(om.SolverWarning, msg):
-            top.run_model()
+        with printoptions(precision=3):
+            msg = (f"'comp.z' exceeds lower bounds\n  Val: [1.333 1.333 1.333]\n  Lower: [1.5 1.5 1.5]\n")
+            with assert_warning(om.SolverWarning, msg):
+                top.run_model()
 
         top.setup()
         # Test upper bounds: should go to the minimum upper bound and stall
@@ -455,9 +455,10 @@ class TestBoundsEnforceLSArrayBounds(unittest.TestCase):
         top['comp.y'] = 0.
         top['comp.z'] = 2.4
 
-        msg = (f"'comp.z' exceeds upper bounds\n  Val: [2.66666667 2.66666667 2.66666667]\n  Upper: [2.6  2.5  2.65]\n")
-        with assert_warning(om.SolverWarning, msg):
-            top.run_model()
+        with printoptions(precision=3):
+            msg = (f"'comp.z' exceeds upper bounds\n  Val: [2.667 2.667 2.667]\n  Upper: [2.6  2.5  2.65]\n")
+            with assert_warning(om.SolverWarning, msg):
+                top.run_model()
 
         for ind in range(3):
             assert_near_equal(top['comp.z'][ind], [2.5], 1e-8)

